README.md
# Mod_Necromant
 Factorio Mod Necromant

Priority:
1 Make this mod work well with Vanilla game
2 Once stable, revisit the optional mod integrations with 5Dim and Bobs


Todo:
* Add migration logic due to renaming of items and recipes
	poison-ammo to ammo-poison

Ideas
* DNA changes #1 - in progress
	More DNA extraction recipes to remove overstock items late stage with by-product stone
	Allow DNA ingredient as part of vanilla research.
* DNA changes #2
	Change for higher DNA extract the by-product from stone to cellulose. 
	Add Cellulose as ingredient to make plastic
* Weapons change - in progress
	Add chitin ammo (upgrade from poison ammo + chitin) 
	Add behemoth ammo (upgrade of chitin ammo + U238)
* Re-balance recipes
	-
* Alien revenge
	Idea: increase alien revenge for butchering their kind.
	Add alien smelters/assemblers/chem plants with higher pollution than vanilla. Pollution factor x 2 - 5?
	Change alien smelters/assemblers/chem plants smoke color to pink
* Cleanup list
	Change alien processing from biter type/size (3x4) to biter size-based research (small, medium, big, behemoth)
* Graphics updates


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.18.7
Date: 2020-03-04
  Changes: 
    - included a quick fix for NE Enemies (disabled by default in mod settings). Original code by regedit200 as posted on https://forums.factorio.com/viewtopic.php?t=37373&start=20
  New:
    - spawners are dropping a corpses
    - added spawner research level 0 research. Changed dependencies.
---------------------------------------------------------------------------------------------------
Version: 0.18.6
Date: 2020-02-26
  Changes: 
    - included a quick fix for NE Enemies (disabled by default in mod settings). Original code by regedit200 as posted on https://forums.factorio.com/viewtopic.php?t=37373&start=20
---------------------------------------------------------------------------------------------------
Version: 0.18.5
Date: 2020-02-25
  Changes: 
    - code cleanup
    - solve bug with DNA from fish/wood
---------------------------------------------------------------------------------------------------
Version: 0.18.4
Date: 2020-02-23
  Changes: 
    - Added migration script
    - Moving Alien DNA recipes out of experimental
    - Minor typo correction, regrouping of icons in menus
    - Added Chitin and Behemoth ammo
---------------------------------------------------------------------------------------------------
Version: 0.18.3
Date: 2020-02-21
  Info:
    - Reformatting change log
  Changes:
    - Fixed the gunshot_sound issues from 0.18.1
    - Adding experimental startup setting.
    - Recipe Alien DNA 1 from Raw fish and Wood   
    - Recipe Alien DNA 2 from Bones, Flesh
    - Recipe Alien DNA 3 from Brain, Steak using protein and acid
    - Recipe Alien DNA 4 from Chitin, Enriched Bones using protein and acid
---------------------------------------------------------------------------------------------------
Version: 0.18.2
Date: 2020-02-17
  Changes:
    - Added behemoth worm to the recipes, dropping 12 bones, 8 flesh and 82 chintin
---------------------------------------------------------------------------------------------------
Version: 0.18.1
Date: 2020-02-17
  Changes:
    - Update by Trblz to make this mod work in 0.18, merging changes from https://mods.factorio.com/mod/necromant-trblz
    - Disabled in prototypes/fireball-turret.lua line 158 sound = make_heavy_gunshot_sounds()
---------------------------------------------------------------------------------------------------
Version: 0.18.0
Date: 2020-02-04
  Info:
    - Quick fix by Trblz in separate mod https://mods.factorio.com/mod/necromant-trblz
---------------------------------------------------------------------------------------------------
Version: 0.17.2
Date: 2019-04-20
  Changes:
    - Fixed one wrong image link for non hr graphics (reported by Thedrah).
---------------------------------------------------------------------------------------------------
Version: 0.17.1
Date: 2019-03-03
  Info:
    - locale correction & added change log
---------------------------------------------------------------------------------------------------
Version: 0.17.0
Date: 2019-03-01
  Info:
    - update for 0.17.x
---------------------------------------------------------------------------------------------------
Version: 0.16.7
Date: 2018-08-19
  Changes:
    - added mod settings for various stuff (DNA, Chitin, Fireball...)
    - added optional dependency 5dim battlefield
    - doubled heal of heal potion
    - reduced amount of ingredients for DNA recipes
---------------------------------------------------------------------------------------------------
Version: 0.16.6
Date: 2018-08-12
  Changes:
    - Added recipe to make alien artifacts from dried alien chemicals.
    - Little technology adjustment due changes.
  Rebalance:
    - Dried alien chemical need now 5 instead of 1 alien chemical.
---------------------------------------------------------------------------------------------------
Version: 0.16.5
Date: 2018-07-14
  Changes:
    - Adjusted recipe for Fireball launcher.
    - Fixed missing description for technology fireball damage.
---------------------------------------------------------------------------------------------------
Version: 0.16.4
Date: 2018-07-07
  Changes:
    - Added fireball tower and fireball artillery.
    - Changed science pack requirement for fireball from blue to military science pack.
    - Chitin processing now requires resin, if bob plates installed.
---------------------------------------------------------------------------------------------------
Version: 0.16.3
Date: 2018-01-12
  Changes:
    - Readded the prerequisite Chitin processing for heavy armor tech.
    - Changed fireball staff to fireball laucher. Compare this https://youtu.be/wvpjAyHH490
    - Fixed the range of the fireball, which became too high in 0.16.x.
    - Reduced the cooldown for the fireball by 25%.
---------------------------------------------------------------------------------------------------
Version: 0.16.2
Date: 2018-01-05
  Info:
    - Added support for 5dim's Battlefield mod.
---------------------------------------------------------------------------------------------------
Version: 0.16.1
Date: 2018-01-04
  Rebalance:
    - Rebalanced fireball. Increased the damage, but reduced the area.
---------------------------------------------------------------------------------------------------
Version: 0.16.0
Date: 2017-12-16
  Info:
    - Upgrade for 0.16.x.
    - Minimal tested, please report the bugs, strange behaviour (fireball may have a problem and need to be improved).
---------------------------------------------------------------------------------------------------
Version: 0.15.9
Date: 2017-11-19
  Changes:
    - Added bob's enemies drops.
    - Code improvemnet for compability.
---------------------------------------------------------------------------------------------------
Version: 0.15.8
Date: 2017-09-30
  Changes:

prototypes/item.lua
data:extend(
{ 
	-- corps parts
	{
		type = "item",
		name = "necro-biter-brain",
		icon = "__necromant__/graphics/icons/biter-brain.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "d",
		stack_size = 200
	},
	-- biter flesh
	{
		type = "item",
		name = "necro-biter-flesh",
		icon = "__necromant__/graphics/icons/flesh.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "e",
		stack_size = 200
	},
	-- alien dna
	{
		type = "item",
		name = "necro-alien-dna",
		icon = "__necromant__/graphics/icons/dna.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "c",
		stack_size = 200
	},
	-- bones
	{
		type = "item",
		name = "necro-bones",
		icon = "__necromant__/graphics/icons/bones.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "i-a",
		stack_size = 200
	},
	-- charcoal
	{
		type = "item",
		name = "necro-bones-charcoal",
		icon = "__necromant__/graphics/icons/bone-charcoal.png",
		icon_size = 32,
		RecipeType = "smelting",
		fuel_category = "chemical",
		fuel_value = "3MJ",
		Enabled = true,
		Recipe = true,
		subgroup = "necromant-filter-products",
		order = "h",
		ResultAmount = 2,
		stack_size = 400
	},
	-- chitin fragment
	{
		type = "item",
		name = "necro-chitin-fragment",
		icon = "__necromant__/graphics/icons/chitin-fragment.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "g",
		stack_size = 1000
	},
	-- chitin
	{
		type = "item",
		name = "necro-chitin",
		icon = "__necromant__/graphics/icons/chitin.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "f",
		stack_size = 200
	},
	-- enriched bones
	{
		type = "item",
		name = "necro-enriched-bones",
		icon = "__necromant__/graphics/icons/enriched-bones.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "i-b",
		stack_size = 200
	},
	-- poison ammo
	{
		type = "ammo",
		name = "necro-ammo-poison",
		icon = "__necromant__/graphics/icons/ammo-poison.png",
		icon_size = 32,
		--subgroup = "ammo",
		subgroup = "necromant-filter-products",
		order = "aa",
		stack_size = 200,
		ammo_type =
		{
		  category = "bullet",
		  action =
		  {
			type = "direct",
			action_delivery =
			{
			  type = "instant",
			  source_effects =
			  {
				  type = "create-explosion",
				  entity_name = "explosion-gunshot"
			  },
			  target_effects =
			  {
				{
				  type = "create-entity",
				  entity_name = "explosion-hit"
				},
				{
				  type = "damage",
				  damage = { amount = 6 , type = "physical"}
				},
				{
				  type = "damage",
				  damage = { amount = 10 , type = "poison"}
				},
				{
				  type = "damage",
				  damage = { amount = 8 , type = "explosion"}
				}
			  }
			}
		  }
		},
		magazine_size = 10,
		order = "z-a",
	},
	-- chitin ammo
	{
		type = "ammo",
		name = "necro-ammo-chitin",
		icon = "__necromant__/graphics/icons/ammo-chitin.png",
		icon_size = 32,
		--subgroup = "ammo",
		subgroup = "necromant-filter-products",
		order = "aa",
		stack_size = 200,
		ammo_type =
		{
		  category = "bullet",
		  action =
		  {
			type = "direct",
			action_delivery =
			{
			  type = "instant",
			  source_effects =
			  {
				  type = "create-explosion",
				  entity_name = "explosion-gunshot"
			  },
			  target_effects =
			  {
				{
				  type = "create-entity",
				  entity_name = "explosion-hit"
				},
				{
				  type = "damage",
				  damage = { amount = 16 , type = "physical"}
				},
				{
				  type = "damage",
				  damage = { amount = 10 , type = "poison"}
				},
				{
				  type = "damage",
				  damage = { amount = 8 , type = "explosion"}
				}
			  }
			}
		  }
		},
		magazine_size = 10,
		order = "z-b",
	},
	-- behemoth ammo
	{
		type = "ammo",
		name = "necro-ammo-behemoth",
		icon = "__necromant__/graphics/icons/ammo-behemoth.png",
		icon_size = 32,
		--subgroup = "ammo",
		subgroup = "necromant-filter-products",
		order = "aa",
		stack_size = 200,
		ammo_type =
		{
		  category = "bullet",
		  action =
		  {
			type = "direct",
			action_delivery =
			{
			  type = "instant",
			  source_effects =
			  {
				  type = "create-explosion",
				  entity_name = "explosion-gunshot"
			  },
			  target_effects =
			  {
				{
				  type = "create-entity",
				  entity_name = "explosion-hit"
				},
				{
				  type = "damage",
				  damage = { amount = 20 , type = "physical"}
				},
				{
				  type = "damage",
				  damage = { amount = 20 , type = "poison"}
				},
				{
				  type = "damage",
				  damage = { amount = 20 , type = "explosion"}
				}
			  }
			}
		  }
		},
		magazine_size = 10,
		order = "z-c",
	},
	-- biter-steak
	{
		type = 'capsule',
		name = 'necro-biter-steak',
		subgroup = "necromant-filter-products",
		icon = '__necromant__/graphics/icons/biter-steak.png',
		icon_size = 32,
		stack_size = 100,
		order = "b",
		capsule_action =
		{
		  type = "use-on-self",
		  attack_parameters =
		  {
			type = "projectile",
			ammo_category = "capsule",
			cooldown = 30,
			range = 0,
			ammo_type =
			{
			  category = "capsule",
			  target_type = "position",
			  action =
			  {
				type = "direct",
				action_delivery =
				{
				  type = "instant",
				  target_effects =
				  {
					type = "damage",
					damage = {type = "physical", amount = -30}
				  }
				}
			  }
			}
		  }
		}
	},
	-- heal
	{
		type = 'capsule',
		name = 'necro-heal-potion',
		subgroup = "necromant-filter-products",
		icon = '__necromant__/graphics/icons/heal-potion.png',
		icon_size = 32,
		stack_size = 100,
		order = "ab",
		capsule_action =
		{
		  type = "use-on-self",
		  attack_parameters =
		  {
			type = "projectile",
			ammo_category = "capsule",
			cooldown = 30,
			range = 0,
			ammo_type =
			{
			  category = "capsule",
			  target_type = "position",
			  action =
			  {
				type = "direct",
				action_delivery =
				{
				  type = "instant",
				  target_effects =
				  {
					type = "damage",
					damage = {type = "physical", amount = -200}
				  }
				}
			  }
			}
		  }
		}
	},
	-- worms
	{
		type = "item",
		name = "necro-worm-chitin",
		icon = "__necromant__/graphics/icons/worms/worm_chitin.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "g",
		stack_size = 200
	},
	-- worm stone wall
	{
		type = "item",
		name = "necro-item-worm-stone-wall",
		icon = "__necromant__/graphics/entity/walls/stone-worm-wall.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "h",
		place_result = "necro-worm-stone-wall",
		stack_size = 200
	},
	-- worm gate
	{
		type = "item",
		name = "necro-item-worm-gate",
		icon = "__necromant__/graphics/icons/stone-worm-gate.png",
		icon_size = 32,
		subgroup = "necromant-filter-products",
		order = "i",
		place_result = "necro-worm-gate",
		stack_size = 200
	},
	-- fire ball
	{
		type = "gun",
		name = "necro-fireball-launcher",
		icon = "__necromant__/graphics/icons/necro-fireball-launcher.png",
		icon_size = 128,
		subgroup = "necromant-filter-magic-products",
		order = "a",
		attack_parameters =
		{
		  type = "projectile",
		  ammo_category = "necro-staff",
		  movement_slow_down_factor = 0.1,
		  cooldown = 15,
		  projectile_creation_distance = 1.0,
		  range = 20,
		  projectile_center = {-0.17, 0},
		},
		stack_size = 5
	},
	-- fireball powder
	{
		type = "ammo",
		name = "necro-fireball-powder",
		icon = "__necromant__/graphics/icons/fireball-powder.png",
		icon_size = 32,
		ammo_type =
		{
		  range_modifier = 3,
		  cooldown_modifier = 3,
		  target_type = "position",
		  category = "necro-staff",
		  action =
		  {
			type = "direct",
			action_delivery =
			{
			  type = "projectile",
			  projectile = "necro-fireball",
			  starting_speed = 0.50,
			  --[[source_effects =
			  {
				type = "create-entity",
				entity_name = "explosion-hit"
			  }
			  --]]
			}
		  }
		},
		subgroup = "necromant-filter-magic-products",
		order = "b",
		stack_size = 1000
	},
	-- burning substance
	{
		type = "item",
		name = "necro-burning_substance",
		icon = "__necromant__/graphics/icons/burning-substance.png",
		icon_size = 32,
		fuel_category = "chemical",
		fuel_value = "5MJ",
		subgroup = "necromant-filter-magic-products",
		order = "c",
		stack_size = 200
	},
	-- dried chemicals
	{
		type = "item",
		name = "necro-dried-chemicals",
		icon = "__necromant__/graphics/icons/dried-alien-chemicals.png",
		icon_size = 32,
		subgroup = "necromant-filter-magic-products",
		order = "e",
		stack_size = 200
	}	
}
)
